### PR TITLE
Support I2C1 on BlueField-2

### PR DESCRIPTION
**Pull request by asmaamellanox**

### CODE REVIEW

File: lanserv/mellanox-bf/mlx-bf.lan.conf [hunk 1]
```diff
@@ -37,5 +37,4 @@ set_working_mc 0x30
   user 1 true  ""        "test" user     10       none md2 md5 straight
   user 2 true  "ADMIN" 	 "ADMIN" admin    10       none md2 md5 straight
 
-  # If IPMB is supported on this system, define SMBus/IPMB channel associated
-  # with channel 2 in OpenIPMI
+  # If IPMB is supported on this system, define SMBus/IPMB channel
```
Comment (dthompso) [inline lanserv/mellanox-bf/mlx-bf.lan.conf:40]: Seems to me like an incomplete sentence.  Is there more you could add to
the comment that helps the reader understand how to define this channel?

File: lanserv/mellanox-bf/set_emu_param.sh [hunk 2]
```diff
@@ -63,21 +59,28 @@ IPMB_HOST_ADD=0x1011
...
-
-is_bluewhale=$1
+bffamily=$1
 support_ipmb=$2
 oob_ip=$3
 
-if [ "$is_bluewhale" = "Bluewhale" ] || [ "$support_ipmb" = "1" ]; then
+if [ "$bffamily" = "Bluewhale" ]; then
+	i2cbus=2
+elif [ "$bffamily" = "BlueSphere" ]; then
```
Comment (dthompso) [inline lanserv/mellanox-bf/set_emu_param.sh:68]: Not related to this code review, but looks like bffamily does not support BlueSphere yet

Comment (asmaamellanox) [reply lanserv/mellanox-bf/set_emu_param.sh:68]: actually I added code to support that. Let me add you as a reviewer.

File: lanserv/mellanox-bf/mlx-bf.lan.conf (hunk 1 above)
Comment (asmaamellanox) [reply lanserv/mellanox-bf/mlx-bf.lan.conf:40]: I will remove this it seems irrelevant to many platforms anyways.

File: lanserv/mellanox-bf/set_emu_param.sh [hunk 3]
```diff
@@ -63,21 +59,28 @@ IPMB_HOST_ADD=0x1011
...
-I2C2_NEW_DEV=/sys/bus/i2c/devices/i2c-2/new_device
-I2C2_DEL_DEV=/sys/bus/i2c/devices/i2c-2/delete_device
-
-is_bluewhale=$1
+bffamily=$1
 support_ipmb=$2
 oob_ip=$3
 
-if [ "$is_bluewhale" = "Bluewhale" ] || [ "$support_ipmb" = "1" ]; then
+if [ "$bffamily" = "Bluewhale" ]; then
```
Comment (dthompso) [inline lanserv/mellanox-bf/set_emu_param.sh:66]: So if the code is running on BlueWhale or BlueSphere, you can't turn off
the IPMB support, right?  Is that ok?  In other words, the support_ipmb
parameter is only consulted if bffamily is not BlueWhale or BlueSphere.

Comment (asmaamellanox) [reply lanserv/mellanox-bf/set_emu_param.sh:66]: yes exactly. IPMB has to be implemented on those 2 platforms. It is a requirement. If you are thinking about debugging scenarios, the user has the ability to disable IPMB even on BW and BSPH using the IPMI config file "progconf". There, we can change the value of the bffamily variable to NONE.

**Review by dthompso — APPROVED** [review]

**Pull request merged**